Client: Improve query params

drvn-mr opened the pull request:
Adds better handling of query parameters using HTTParty instead of manually.
Fills in the gaps in the supported WeatherKit query params.

Note: I should check if `weather_query_for_options` is still valid.
Old code.

woodie commented:
Yes, and it is very difficult to mock the current implementation in specs. 

Please consider compacting the options.

drvn-mr commented:
The query is only compacted when going through the passed options.
Updated tests.

woodie commented:
This is a great change. I started adding some additional specs [here](https://github.com/superbasicxyz/tenkit/pull/31/files). Feel feel to pull them over or do something else. :-)

I suppose the helper methods could to more to avoid this: `[Tenkit::Utils.camel(k.to_s).to_sym`

Let me know if you need me to rebase (or whatever). @drvn-mr

drvn-mr commented:
It would be much easier to just wait for these simple changes to be merged first.
You could improve using helpers afterwards.

drvn-mr committed:
Client: Improve query params

Adds better handling of query parameters using HTTParty instead of manually.
Fills in the gaps in the supported WeatherKit query params.
